app/renderer/js/utils/dnd-util.js

```diff
@@ -2,30 +2,168 @@
 
 const ConfigUtil = require(__dirname + '/config-util.js');
 
+const {ipcRenderer} = require('electron');
+
+let count = 0;
+let	dnd;
+let newSettings;
+let time = -1;
+function makeView() {
+	const dndOffTime = {
+		hr: new Date().getHours(),
+		min: new Date().getMinutes(),
+		carry: 0, // for dnd switch on after midnight 0 am
+		day: new Date().getDay()
+	};
+	const timeLeft = document.createElement('span');
+	const timeDisableDNDInput = document.createElement('div');
+	const actionContainer = document.getElementById('actions-container');
+	timeLeft.id = 'timeLeft';
+	actionContainer.prepend(timeLeft);
+	timeDisableDNDInput.className = 'dndInput';
+	for (let i = 1; i <= 4; i++) {
+		const opt = document.createElement('span');
+		if (i === 1) {
+			opt.innerHTML = 'For 1 hour<br/>';
+			opt.id = 'dnd_1';
+		} else if (i === 2) {
+			opt.innerHTML = 'For 8 hour<br/>';
+			opt.id = 'dnd_2';
+		} else if (i === 3) {
+			opt.innerHTML = 'For 12 hour<br/>';
+			opt.id = 'dnd_3';
+		} else if (i === 4) {
+			opt.innerHTML = 'Until I resume<br/>';
+			opt.id = 'dnd_4';
+		}
+		opt.addEventListener('click', () => {
+			element(opt, dndOffTime, timeDisableDNDInput);
+		}, false);
+		timeDisableDNDInput.appendChild(opt);
+		actionContainer.appendChild(timeDisableDNDInput);
+	}
+}
+
+function element(opt, dndOffTime, timeDisableDNDInput) {
+	const optionElement = document.getElementById(opt.id);
+	const actionContainer = document.getElementById('actions-container');
+	optionElement.className = 'dndOptionsSelect';
+	if (opt.id === 'dnd_1') {
+		time = 1;
+	} else if (opt.id === 'dnd_2') {
+		time = 8;
+	} else if (opt.id === 'dnd_3') {
+		time = 12;
+	} else {
+		time = -1;
+	}
+	if (time > 0) {
```

Comment:
The `dndOffTime` could be somewhere here.

Reply:
sure. The some variable declarations have been misplaced with the improve of code from the first check, when the CI tests failed, in order to satisfy the pre-commit conditions.